src/lib/utils/common-helper.ts

```diff
@@ -71,10 +72,57 @@ function exhaustiveCheck(x: never): never {
  * @param count - the number of items
  * @param singular - the singular form of the word
  * @param plural - the plural form of the word
- * @returns the singular form if count is 1, otherwise the plural form
+ * @return the singular form if count is 1, otherwise the plural form
  */
 function pluralize(count: number, singular: string, plural: string): string {
     return count === 1 ? singular : plural;
 }
 
-export { getName, getNames, isPaperUndecided, doesPaperNeedReview, exhaustiveCheck, pluralize };
+/**
+ * Groups a list by keys.
+ * The keys are indirectly given by the keySelector function.
+ *
+ * Inspired by: https://stackoverflow.com/questions/14446511/most-efficient-method-to-groupby-on-an-array-of-objects
+ *
+ * @param list the list to be grouped
+ * @param keySelector function that map a (list) item to a certain key
+ * @return the grouped list as an object with the association key: \<list of items associated to key\>
+ */
+function groupBy<T>(list: T[], keySelector: (arg0: T) => string): { [key: string]: T[] } {
+    return list.reduce((result: { [key: string]: T[] }, item: T) => {
+        (result[keySelector(item)] ??= []).push(item);
+        return result;
+    }, {});
+}
```

Comment:
Use the `Record` type for such cases.
```suggestion
function groupBy<T>(list: T[], keySelector: (arg0: T) => string): Record<string, T[]> {
    return list.reduce((result: Record<string, T[]>, item: T) => {
        (result[keySelector(item)] ??= []).push(item);
        return result;
    }, {});
}
```

Reply:
Good idea, didn't think of it.